File: vb.net/Safadometro/Safadao.vb
Imports System.Linq

Public Class Safadao
    Public Dim Ano As Integer
    Public Dim Mes As Integer
    Public Dim Dia As Integer

    Public Dim Safadeza As Double
    Public Dim Anjo As Double

    Sub New(dia As Integer, mes As Integer, ano As Integer)
        Me.Dia = dia
        Me.Mes = mes
        Me.Ano = ano

        Me.Safadeza = CalcSafadeza()
        Me.Anjo = CalcAnjo()
    End Sub

    Private Function SumMonth() As Integer
        return Enumerable.Range(1, Me.Mes).Sum()
    End Function

    Private Function CalcAnjo() As Double
        Return 100 - Me.Safadeza
    End Function

    Private Function CalcSafadeza() As Double
        Return SumMonth() + (Me.Ano / 100d) * (50 - Me.Dia)
    End Function

    Sub SaySafadeza()
        Console.WriteLine(String.Format("Você é {0}% safado e {1}% anjo", Safadeza, Anjo))
    End Sub
End Class

File: erlang/safadao.erl
-module(safadao).
-export([safadeza/3]).


safadeza(Dia, Mes, Ano) ->
	safadeza(Dia, Mes, Ano, 0).


safadeza(Dia, 0, Ano, FatorSafadao) -> 
	AnoSafado = Ano div 100,
	DiaSafado = 50 - Dia,
	Safadeza = FatorSafadao + (AnoSafado * DiaSafado),
	Anjo = 100 - Safadeza,
	io:format("~p% anjo, perfeito, mas aquele ~p% ... \n", [ Anjo, Safadeza ] );

safadeza(Dia, Mes, Ano, FatorSafadao ) ->	
	safadeza(Dia, Mes - 1, Ano, FatorSafadao + Mes ).
	

	
